api/v1alpha3/types.go

```diff
@@ -138,6 +138,12 @@ type VirtualMachineCloneSpec struct {
 	// Defaults to empty map
 	// +optional
 	CustomVMXKeys map[string]string `json:"customVMXKeys,omitempty"`
+	// ProvisionerType is the name of provisioner type to use. Can be either cloud-init or
```

Comment:
With https://github.com/kubernetes-sigs/cluster-api/pull/4172, this would not be needed and could be auto-detected from `"format"` field in bootstrap secret.

Reply:
Cool, reading this from the secret field is a better option. This PR was opened before your changes.

Reply:
There was a suggestion in this PR, that maybe it should be stored in annotation, but it's an open case for now. In any case, it's a minor change anyway.